lib/SILOptimizer/Transforms/DeadObjectElimination.cpp

```diff
@@ -224,6 +269,28 @@ static bool canZapInstruction(SILInstruction *Inst, bool acceptRefCountInsts) {
   if (isa<DestroyAddrInst>(Inst))
     return true;
 
+  // As a special rule, a string interpolation is eliminatable if its result is
+  // unused, even though this theoretically could remove the side effects of
+  // StringInterpolationProtocol methods or the string-conversion witnesses they
+  // would have called if not eliminated. Mutating methods this rule applies to
+  // should be marked with @_semantic("interpolation.append"); they will only
+  // allow their self parameter to be eliminated.
+  if (auto *Apply = dyn_cast<ApplyInst>(Inst))
+    if (appendsToAllocatedInterpolation(Apply, AllocInst))
+      return true;
+
+  // We can remove loads of addresses within the allocation.
+  if (auto *Load = dyn_cast<LoadInst>(Inst))
+    return getBaseAddrInst(Load) == AllocInst &&
+            // HACK: In the general case, we should find any memory management
+            // done with the results of this load, then find any corresponding
+            // stores into the allocation and replace the stores with equivalent
+            // memory management. We don't do that here (that's what
+            // DeadObjectAnalysis below is supposed to do), but we know we don't
+            // need to for DefaultStringInterpolation, because we know that the
+            // initialization of its sole property will be eliminated too.
```

Comment:
I don't understand this comment. Can you give an example?

Reply:
I may be a little confused here. What I'm trying to say is that in the general case, if we eliminate a load and a release, we should look for corresponding stores and replace them with releases of the values that would have been stored into them. We don't need to do that for `DefaultStringInterpolation` because we know that if we eliminate its allocation, we will also eliminate anything that could have allocated or retained the string storage object that would have been inside it, because that only happens inside eliminatable calls to `appendLiteral(_:)`, `appendInterpolation(_:)`, or `init(literalCapacity:interpolationCount:)`.

Reply:
I see. Can you change the comment to this explanation?